enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +669,220 @@ guarantee uniqueness and eliminates the ability to falsify access.
+Having the prefix avoids conflicting with existing NADs who has the same `metadata.name`.
+
+Creating cluster scoped CRD instance should trigger creation of the corresponding NAD at each namespace specified in the spec.
+Following the above cluster-scope CRD [example](#example---cluster-scoped-network), the following NADs should be created:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: shared-db-network          <--- name starts with "shared"
+  namespace: mynamespace
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: ClusterUserDefinedNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4 
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"shared-db-network",   <--- generated unique network-name
+          "netAttachDefName":"mynamespace/db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "role": "primary"
+      }'
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: shared-db-network                 <--- same name as in other nameapces
+  namespace: theirnamespace
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  ownerReferences:
+  - apiVersion: k8s.ovn.org/v1alpha1
+    blockOwnerDeletion: true
+    kind: ClusterUserDefinedNetwork
+    name: db-network
+    uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >
+    '{
+        "cniVersion":"0.3.1",
+        "excludeSubnets":"10.0.0.100/24",
+        "mtu":1500,   
+        "name":"shared-db-network",   <--- same name as in other namespaces
+        "netAttachDefName":"theirnamespace/db-network",
+        "subnets":"10.0.0.0/24",
+        "topology":"layer2",
+        "type":"ovn-k8s-cni-overlay",
+        "role": "primary"
+    }'
+```
+
+##### Validations
+The controller should validate the request CRD spec and verify:
+- CIDRs are valid.
+- MTU is valid, no more than 9000.
+- `Topology` is one of `Layer2`, `Layer3` or `Localnet`.
+- `Role` is one of `Primary` or `Secondary`.
+- `IPAMLifecycle` can be `Persistent`, and set only when topology is `Layer2` or `Localnet`. 
+- In case `Topology: Localnet`, `Role` cannot be `Primary`
+
+In addition, the following scenarios should be validated: 
+- The join subnet shall not overlap with the configured cluster default network join subnet.
+  If there is overlap, OVN-Kubernetes should report an error in the request CR status.
+
+- When primary network is requested (i.e.: `spec.role: Primary`)
+  - Verify no primary network exist at the target namespace (i.e.: no NAD with `"primaryNetwork": "true"` exist).
```

Comment:
I think we can only guarantee that if NADs are not created by cluster admin directly. We still need to figure out what to do when primary network UDN is created, then admin creates primary network NAD.
Since NAD doesn't have a status and we watch all NADs anyway, we could try to report this problem on conflicting NAD create via UDN status, saying that UDN is fine, but there is a conflicting NAD that needs to be removed.
It is a bit racy at this point: if NAD was created first, UDN won't create its own NAD, but if UDN was created first, we won't delete UDN's NAD. I think this is fine as long as we report these details to the user.

Reply:
Done
Added text to validation section about the described above scenario.

FWIW please note the `Best practices for managing OVN network using CRDs` refers to this topic, it suggest to not create more then one primary network in a namespace 
I added more text to clearly it .

Reply:
I think what you propose is fine @npinaeva. I don't want to worry to much about the "what ifs" with an admin bypassing the CRD and creating conflicting NADs. As we discussed in our stand up meetings, the admin is at their own risk if they start tinkering with that.